### ISSUE TITLE
Measures extending past margin of page

### ISSUE THREAD

**Issue by ShaiMartin** [musescore/MuseScore#20773]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

In two of my part scores, I have measures extending past the initial margin of the page, which I can't click on to edit (picture attached). Both of these only apply when a certain system is visible. In my piano part-score, if I make the left hand invisible, it aligns normally. In the guitar part-score, in which I have a notation and a tab part in, it aligns normally if I hide the tab part.

### Steps to reproduce

A zipped MSCZ file is attached, the bug is evident in the guitar and piano part scores.
[Score with bug in piano and guitar part scores.mscz.zip](https://github.com/musescore/MuseScore/files/13799368/Score.with.bug.in.piano.and.guitar.part.scores.mscz.zip)

### MuseScore Version

OS: macOS 11.7, Arch.: x86_64, MuseScore version (64-bit): 4.2.0-233521125, revision: github-musescore-musescore-eb8d33c

### Regression

OK in 4.0, broken in 4.1

### Operating system

OS: macOS 11.7

### Additional context

![Screen Shot 2023-12-19 at 9 52 30 am](https://github.com/musescore/MuseScore/assets/154406050/916f7a82-b353-444e-b77c-2ca2222d7edc)
![Screen Shot 2023-12-19 at 9 52 42 am](https://github.com/musescore/MuseScore/assets/154406050/0a6ce8b1-2a19-4dbf-86e9-db2e18d49ec5)

**Comment by cadiz2** [comment]
Initial discussion on forum: https://musescore.org/en/node/357957

**Comment by mike-spa** [comment]
Fixed by #21210. In the meantime @ShaiMartin the solution is to do as suggested by @MarcSabatella: right click any of the Layout Break marks in tha page (e.g. the one between bars 7-8), do "Select similar", and delete all of them. 

@bkunda @oktophonie I think there is a case to be made here that the use of "Keep measures on the same system" should be discouraged as much as possible (not least because we want to get rid of it at some point). Maybe writing a warning in the handbook, if we don't already? (I think we do something similar for a few other things too). 
<img width="235" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/6498f5a6-bcb8-43a4-b18c-1fbc7d0a5ae3">
I did put in some work last year to make it behave somewhat more sensibly, but it won't fix the basic fact that using this item inappropriately can force the spacing system to do something that it isn't design for. This causes bugs such as this (and indeed https://github.com/musescore/MuseScore/issues/20995) but also has a very signifcant performance costs. 

This item should only really be used for keeping 2-measure-repeats together (that's the reason it was origiginally introduced afaik).
